docs/src/tutorial/pure.rst

```diff
@@ -328,15 +328,27 @@ declare types of variables in a Python 3.6 compatible way as follows:
 
 There is currently no way to express the visibility of object attributes.
 
-Cython does not support the full range of annotations described by PEP-484.
-For example it does not currently understand features from the ``typing`` module
-such  as ``Optional[]`` or typed containers such as ``List[str]``. This is partly
-because some of these type hints are not relevant for the compilation to
+``typing`` Module
+^^^^^^^^^^^^^^^^^
+
+Support for the full range of annotations described by PEP-484 is not yet
+complete. Cython 3 currently understands the following features from the
+``typing`` module:
+
+* ``Optional[tp]``, which is interpreted as ``tp or None``;
+* typed containers such as ``List[str]``, which is interpreted as ``list``. The
+  hint that the elements are of type ``str`` is currently ignored;
```

Comment:
I am not that experienced in how Cython works yet, is there actually anything useful that Cython can do with these typings?

Reply:
> I am not that experienced in how Cython works yet, is there actually anything useful that Cython can do with these typings?


There's small speedups for knowing that something's a list `list` or `dict` instead of an object. Right now there's nothing more Cython could do with "list of strings" over "list" and that's probably unlikely to change.